src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1064,6 +1089,27 @@ LedgerManagerImpl::advanceLedgerPointers(LedgerHeader const& header,
     mLastClosedLedger.header = header;
 }
 
+void
+LedgerManagerImpl::maybeUpdateNetworkConfig(bool upgradeHappened,
+                                            AbstractLedgerTxn& rootLtx)
+{
+    if (!upgradeHappened && mLastContractNetworkConfig)
+    {
+        return;
+    }
+    if (!mLastContractNetworkConfig)
+    {
+        mLastContractNetworkConfig =
+            std::make_optional<ContractNetworkConfig>();
+    }
+    LedgerTxn ltx(rootLtx, false, TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
```

Comment:
This line is making me think that the LedgerTxns added in the [config upgrade PR](https://github.com/stellar/stellar-core/pull/3674/files) should be using `READ_ONLY_WITHOUT_SQL_TXN` since they should all be read only. I'm not sure yet if there are concurrency issues with this, but I forgot this option existed for `LedgerTxn` when I made those changes. What do you think?

1. https://github.com/stellar/stellar-core/blob/bff2c2b37402acc61166c0f369da1205efbe8c9d/src/herder/HerderImpl.cpp#L1210
2. https://github.com/stellar/stellar-core/blob/bff2c2b37402acc61166c0f369da1205efbe8c9d/src/herder/HerderImpl.cpp#L1278
3. https://github.com/stellar/stellar-core/blob/bff2c2b37402acc61166c0f369da1205efbe8c9d/src/herder/Upgrades.cpp#L632

Reply:
Updated.

Reply:
Is there a difference between passing in true vs false for `shouldUpdateLastModified` when `TransactionMode == READ_ONLY_WITHOUT_SQL_TXN`? I don't think there is, but we pass in true for the current use cases, while the fee config PR passes in false.

Reply:
Also, I opened a PR related to using `READ_ONLY_WITHOUT_SQL_TXN` - https://github.com/stellar/stellar-core/issues/3716.